test/release/examples/primers/specialMethods.chpl

```diff
@@ -0,0 +1,173 @@
+// Special Methods
+
+/* This primer covers special methods for classes and records */
+
+// Classes and records can have a handful of methods that are specially named
+// and treated differently than other methods.  These methods enable
+// operations such as initializing or deinitializing a class or record
+// instance, accessing a class or record as though it were an array,
+// iterating over an class or record, and defining the way a class or record
+// is read from or written to a channel.
+//
+// Throughout this primer, the methods are defined as secondary methods
+// using the form: ``proc TypeName.methodName() ...``
+//
+// It is also possible to declare these methods within the body of the
+// class or record as primary methods: ``proc methodName() ...``
+
+// First we will declare a simple record with a field that is a tuple of
+// integers.  We'll add special methods and iterators to this record later.
+record R {
+  param size: int;
+  var vals: size*int;
+}
+
+/*
+  Initializers and Deinitializers
+  -------------------------------
+*/
+
+// An initializer is called when creating an instance of the class or record,
```

Comment:
Isn't there some documentation on initializers that this should refer to? Maybe classes.chpl and records.chpl ?

I think it's important to call out the special initializers:
 * the default initializer
 * the copy initializer

even if you don't describe them in detail (you could just say they exist and refer to the other parts - up to you).

Reply:
I'd probably prefer a strategy of pointing out that default init / copy init / deinit are a kind of special methods (and giving a 1 sentence or 1 phrase summary of each, no code sample) and then refer to other documentation for the details.